### PR TITLE
zynq7000/console: Use UART pin definitions from board_config.h

### PR DESCRIPTION
**Pull request by MaciejPurski**
JIRA: RTOS-203

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (armv7a9-zynq7000-zturn).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by agkaminski — DISMISSED** [review]

File: hal/armv7a/zynq7000/zynq.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * %LICENSE%
+ */
+
+#ifndef _HAL_ZYNQ_H_
+#define _HAL_ZYNQ_H_
+
+#include "../../cpu.h"
+
+
+int _zynq_setMIO(unsigned int pin, char disableRcvr, char pullup, char ioType, char speed, char l0, char l1, char l2, char l3, char triEnable);
```
Comment (HBuczynski) [inline hal/armv7a/zynq7000/zynq.h:22]: Missing `extern`.

File: hal/armv7a/zynq7000/zynq.h [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _HAL_ZYNQ_H_
+#define _HAL_ZYNQ_H_
+
+#include "../../cpu.h"
```
Comment (HBuczynski) [inline hal/armv7a/zynq7000/zynq.h:19]: Probably, it is not needed.

File: hal/armv7a/zynq7000/zynq.c [hunk 3]
```diff
@@ -595,32 +595,7 @@ static void _zynq_peripherals(void)
...
-
-
-	/* Define UARTs' clocks speed
-	 * IO_PLL / 20 :  1000 MHz / 20 = 50 MHz
-	 * CLKACT0 = 0x0; CLKACT1 = 0x1; SRCSEL = 0x0; DIVISOR = 0x14 */
-	*(zynq_common.slcr + slcr_uart_clk_ctrl) = (*(zynq_common.slcr + slcr_uart_clk_ctrl) & ~0x00003f33) | 0x00001402 | 0x1;
-
-	/* Enable UART_0 & UART_1 clock */
+	/* Enable UART_0 & UART_1 clock simultaneously to avoid race condition */
 	*(zynq_common.slcr + slcr_aper_clk_ctrl) = *(zynq_common.slcr + slcr_aper_clk_ctrl) | (1 << 21) | (1 << 20);
```
Comment (HBuczynski) [inline hal/armv7a/zynq7000/zynq.c:599]: You can make `_zynq_setAmbaClk` as a part of the interface, similar to the `_zynq_setMIO` and `_zynq_peripherals` would be removed. Then everything related to uart will be done in `console`.

Comment (MaciejPurski) [reply hal/armv7a/zynq7000/zynq.c:599]: I was considering using `_zynq_setAmbaClk` here, however this function sets only one clock at a time. From, what I understood from @agkaminski, we want to set both clocks simultaneously with just one write, so I left it like it is.

File: hal/armv7a/zynq7000/zynq.h (hunk 1 above)
Comment (MaciejPurski) [reply hal/armv7a/zynq7000/zynq.h:22]: I know we use `extern` in many places, but not everywhere and also it has no effect here.

Comment (HBuczynski) [reply hal/armv7a/zynq7000/zynq.h:22]: Take a look on other kernel modules. In the majority of headers there is an `extern` keyword. I think that code should be consistent.

**Pull request merged**